datafusion/common/src/config.rs

```diff
@@ -1564,7 +1564,7 @@ config_namespace_with_hashmap! {
 config_namespace! {
     /// Options controlling CSV format
     pub struct CsvOptions {
-        pub has_header: bool, default = true
+        pub has_header: Option<bool>, default = None
```

Comment:
👍

Reply:
I think it would help to explain why this is an `Option` which is different than the other options. Something like

```suggestion
        /// If the first line of the CSV is column names. 
        /// If not specified, uses default from `CREATE TABLE` command, if any. 
        pub has_header: Option<bool>, default = None
```